Week-3/exercise-3.1/memoize.js

```diff
@@ -0,0 +1,40 @@
+// Exercise 3.1
+// Create a memoize function that remembers previous inputs and stores them in cache so that it won’t have
+// to compute the same inputs more than once. The function will take an unspecified number of integer inputs
+// and a reducer method.
+
+const memoize = (fn) => {
+  const cache = new Map();
+  return (...args) => {
+    let key = args.toString();
```

Comment:
this approach will fail if the same arguements are sent in other order
e.g add(100,200) and add(200,100) are the same, but the add(200,100) will not use the memoized value

how will you solve that?